stream/src/main/scala/org/apache/pekko/stream/impl/Unfold.scala

```diff
@@ -32,7 +33,7 @@ import pekko.stream.stage.{ GraphStage, GraphStageLogic, OutHandler }
 @InternalApi private[pekko] final class Unfold[S, E](s: S, f: S => Option[(S, E)]) extends GraphStage[SourceShape[E]] {
   val out: Outlet[E] = Outlet("Unfold.out")
   override val shape: SourceShape[E] = SourceShape(out)
-  override def initialAttributes: Attributes = DefaultAttributes.unfold
+  override def initialAttributes: Attributes = DefaultAttributes.unfold and SourceLocation.forLambda(f)
```

Comment:
may I ask what is the purpose and effect of adding `SourceLocation.forLambda(f)` here?